python/cudaq/kernel/ast_bridge.py

```diff
@@ -390,6 +393,7 @@ def promoteOperandType(self, ty, operand):
                 imag = self.getConstantFloatWithType(0.0, floatType)
                 operand = complex.CreateOp(complexType, real, imag).result
 
+        #FIXME: use cc.cast for all below
```

Comment:
+1

Reply:
I will do it in a follow-up PR as I expect some test changes